### PR TITLE
Wrap errors of modules' keeper, querier and msg to sent them to clients

### PR DESCRIPTION
**Pull request by NicolasMahe**
closes https://github.com/mesg-foundation/engine/issues/1711

Wrap errors of modules' keeper, querier and msg with the `sdkerrors.Wrap` function.
This allows the error's message to be sent to clients.
Before:
```
➜  engine git:(feature/wrap-errors) ✗ curl localhost:1317/service/get/QY8K2bPZMx3UxbU4ykVGBM6JYKf2B4ztgGP2igQG9Sb
{"error":"internal error"}%
```
After:
```
➜  engine git:(feature/wrap-errors) curl localhost:1317/service/get/QY8K2bPZMx3UxbU4ykVGBM6JYKf2B4ztgGP2igQG9Sb
{"error":"invalid request: service \"QY8K2bPZMx3UxbU4ykVGBM6JYKf2B4ztgGP2igQG9Sb\" not found"}%
```

### CODE REVIEW

File: x/execution/handler.go [hunk 1]
```diff
@@ -30,7 +26,7 @@ func NewHandler(k Keeper) sdk.Handler {
 func handleMsgCreateExecution(ctx sdk.Context, k Keeper, msg MsgCreateExecution) (*sdk.Result, error) {
 	exec, err := k.Create(ctx, msg)
 	if err != nil {
-		return nil, sdkerrors.Wrap(errors.ErrUnknown, err.Error())
+		return nil, err
```
Comment (antho1404) [inline x/execution/handler.go]: why removing them here?

File: x/ownership/internal/types/msg.go [hunk 2]
```diff
@@ -35,13 +35,13 @@ func (msg MsgWithdrawCoins) Type() string {
 // ValidateBasic runs stateless checks on the message.
 func (msg MsgWithdrawCoins) ValidateBasic() error {
 	if msg.Amount.IsAnyNegative() {
-		return sdkerrors.Wrap(sdkerrors.ErrInsufficientFunds, "price must be positive")
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, "price must be positive")
 	}
 	if msg.Owner.Empty() {
-		return sdkerrors.Wrap(sdkerrors.ErrInvalidAddress, msg.Owner.String())
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, msg.Owner.String())
```
Comment (antho1404) [inline x/ownership/internal/types/msg.go]: Why changing the invalid address error? It seems that this error give less context

File: x/execution/handler.go (hunk 1 above)
Comment (NicolasMahe) [reply x/execution/handler.go]: before the error is return from a keeper function that should always already returned a wrap error.

File: x/ownership/internal/types/msg.go (hunk 2 above)
Comment (NicolasMahe) [reply x/ownership/internal/types/msg.go]: For simplicity I changed most of the error to `ErrInvalidRequest` but maybe that was too much.
I will take a look to see which error can be used to give more context.

**Comment by NicolasMahe** [comment]
TODO:
- [ ] use better error
- [ ] create error per module like `github.com/cosmos/cosmos-sdk/x/gov/types/msgs.go` or create app-wide errors?

File: x/execution/handler.go (hunk 1 above)
Comment (antho1404) [reply x/execution/handler.go]: It's a bit weird to have some errors that are returned as error that don't need wrapping and some that need. Would love to find a way to fix that, maybe a specific check might be good or a specific type.

Anyway, this is not blocking and can be improved later

Comment (NicolasMahe) [reply x/execution/handler.go]: I checked to see if I could check if an error is already wrapped but couldn't find a nice way.

File: x/ownership/internal/types/msg.go (hunk 2 above)
Comment (NicolasMahe) [reply x/ownership/internal/types/msg.go]: I will improve the error on msg in a next PR when I improve the msg fields ;)

**Comment by NicolasMahe** [comment]
@antho1404 ready to review

File: x/execution/internal/keeper/keeper.go [hunk 3]
```diff
@@ -337,9 +340,6 @@ func (k *Keeper) distributePriceShares(ctx sdk.Context, execAddress, runnerAddre
 		if err != nil {
 			return err
 		}
-		if err != nil {
```
Comment (antho1404) [inline x/execution/internal/keeper/keeper.go]: 🤣 good to make sure that we return the error

**Pull request merged**